pokemon/src/contexts/Pokemon/index.tsx

```diff
@@ -0,0 +1,98 @@
+import type { FC } from 'react';
+import { createContext, useContext, useEffect, useState } from 'react';
+import { PokemonContextProps, PokemonContextValue } from './types';
+import { useInfiniteQuery, useQuery } from '@tanstack/react-query';
+import {
+  getPokemonClient,
+  AppPkmnDetail,
+  getAppPkmnDetailFromApi,
+} from '@helpers/index';
+
+const PokemonContext = createContext<PokemonContextValue | undefined>(
+  undefined
+);
+
+const PokemonProvider: FC<PokemonContextProps> = ({ children }) => {
+  const client = getPokemonClient();
+  const [selectedName, setSelectedName] = useState('');
+  const [selected, setSelected] = useState<AppPkmnDetail | undefined>(
+    undefined
+  );
+
+  const fetchPokemons = async ({ pageParam }: { pageParam: string }) => {
+    const offset = Number(pageParam);
+    const res = await client.listPokemons(offset, offset === 100 ? 51 : 50);
+    return res;
+  };
+
+  const { data, isFetching, isFetchingNextPage, fetchNextPage } =
+    useInfiniteQuery({
```

Comment:
`useInfiniteQuery` implementation seems incomplete, currently relies on useEffect in App to fetch remaining pages, combined with hardcoded offset in `await client.listPokemons(offset, offset === 100 ? 51 : 50);`

Reply:
Yeah. This is just because of the exercise's setting. I could've just not used the infinite query here, but my idea was just to have it "ready" for an infinite scrolling list if it would be implemented - sort of just to show that I know how that goes.